### PR TITLE
Update to ungoogled-chromium 127.0.6533.88-1

### PR DESCRIPTION
**Pull request by teeminus**
Builds and runs fine, only two minor adaptions of the build script required to adapt to changes of the download script.

image

### CODE REVIEW

**Comment by teeminus** [comment]
Dang, I found a bug in the clone script which mysteriously disappeared yesterday evening. I am currently investigating a fix for that.

**Comment by teeminus** [comment]
Turns out the clone script uses a more recent version of `gn` which is not yet supported. I added another step to the build script to setup the correct `gn` version.

Besides that, building with the tarball is currently broken. This is caused by an incomplete pruning list in the core repo which results in both the linux and windows of `rustc` being present which causes the chromium scripts to use the linux binary on windows - which obviously fails.

@networkException @Nifury @PF4Public I would like to raise the question if we want to continue to support tarballs at all. The clone script has proven itself while the google tarbot failes on a regular base. And they can stop providing tarballs anytime.

**Comment by teeminus** [comment]
> "We" as in the Windows packaging?

Yes. I don't know which other platforms still use the tarball.

**Comment by networkException** [comment]
Most I think. Alright, I think it would be great to see a platform fully adopt from source checkout

**Comment by teeminus** [comment]
BTW, if we agree on removing the tarball support I would do that for the next major release. I think that's nothing we should do on minor releases.

**Comment by PF4Public** [comment]
> I would like to raise the question if we want to continue to support tarballs at all.

You and @Nifury maintain this repo, so if it would be for you easier to use a clone script instead of supporting tarballs, I'd say go for it. We're all volunteering here, so if something makes it easier on you — should be "good enough". 

My least(!) significant concern might be how could it impact the end-users building themselves? If that's just replacing a wget or curl command with another one, all other things being unchanged, then it shouldn't be a problem at all. Another thing to consider: would using clone script increase the space/bandwidth requirements for Github Actions for us to build binaries?

> And they can stop providing tarballs anytime.

IIRC they tried once, but got a huge backlash from Linux distros, so they reverted it. As long as they provide tarballs for Linux, I'll be using them on Gentoo for their benefits.

**Comment by teeminus** [comment]
We are actually using the clone script as the default for quite some time now. It's faster and requires less storage and bandwidth. The tarball is just a fallback solution right now.

The problem with the tarball is, that the pruning list from the core repo does not work anymore. I think the pruning list there is generated from the clone script as well. Since we switched to the clone script we are providing our own pruning list for windows as the clone script downloads platform specific binaries which are not pruned otherwise.

File: build.py [hunk 1]
```diff
@@ -172,11 +172,29 @@ def main():
 
             # Unpack chromium tarball
             get_logger().info('Unpacking chromium tarball...')
-            downloads.unpack_downloads(download_info, downloads_cache, source_tree, extractors)
+            downloads.unpack_downloads(download_info, downloads_cache, source_tree, None, extractors)
         else:
             # Clone sources
             subprocess.run([sys.executable, str(Path('ungoogled-chromium', 'utils', 'clone.py')), '-o', 'build\src', '-p', 'win32' if args.x86 else 'win64'], check=True)
```
Comment (Nifury) [inline build.py:178]: We might want to escape the backslash in `'build\\src'`. 

Also, we could bump the Python version requirement to 3.12. It builds and runs fine on my end. 

And yeah I'm good with dropping tarball support if the tarball requires more work.

Comment (teeminus) [reply build.py]: You are right, I fixed that.

**Comment by teeminus** [comment]
The advantage of the tarball is, that you have do download it only once. I think the clone script has a build-in detection if the source folder has uncomitted changes and reverts those changes, but if you delete the source folder you have to re-download the chromium sources. But thats only an issue when updating to a new release, this should not affect the average user.

For now, I don't have a solution to make the current tarball work again. But the issue with the pruning list might affect other distros as well, so it might get fixed anyways.

In fact, using the tarball is the only option to compile any release since we switched to the clone script as the clone script always uses the latest `gn` version available which is not compatible with the chromium sources (yet). There were no changes to `gn` for quite some time, that's why we were unaware of this issue. And there might be similar issues that are currently unknown that might break the build process in the future.

I could add a user interaction or at least a note to the build script that building using the tarball is currently broken.

I'm not sure if we can use Python 3.12 without changing at least one patch. If I remember correctly, there was a deprecation warning showing up in the logs, but I am not quite sure. Are there any advantages of using Python 3.12? Do we have to update the CI container as well?

**Review by Nifury — APPROVED** [review]
There are no obvious advantages of using Python 3.12 AFAIK. The clone script used to fail on 3.12, but it doesn't anymore. Just saying we could update the Python version check in `build.py` and README.

**Pull request merged**